apps/mobile/src/libs/components/task-card/task-card.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import { TaskStatus } from "shared";
+
+import {
+	Deadline,
+	Icon,
+	Tag,
+	TaskActionButton,
+	Text,
+	View,
+} from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import { useCallback } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import { type TaskDto } from "~/packages/tasks/tasks";
+
+import { styles } from "./styles";
+
+type Properties = {
+	onComplete?: (id: number) => void;
+	onSkip?: (id: number) => void;
+	task: TaskDto;
+};
+
+const EDIT_ICON_SIZE = 20;
+
+const TaskCard: React.FC<Properties> = ({ onComplete, onSkip, task }) => {
+	const handleTaskSkip = useCallback(() => {
+		onSkip?.(task.id);
+	}, [task, onSkip]);
+
+	const handleTaskComplete = useCallback(() => {
+		onComplete?.(task.id);
+	}, [task, onComplete]);
+
+	const isActiveTask = task.status === TaskStatus.CURRENT;
+
+	const getButtonProperties = (): {
+		label: string;
+		type: "complete" | "skip";
+	} | null => {
+		if (task.status === TaskStatus.SKIPPED) {
+			return { label: "Skipped", type: "skip" };
+		}
+
+		if (task.status === TaskStatus.COMPLETED) {
+			return { label: "Completed", type: "complete" };
+		}
+
+		return null;
+	};
+
+	const buttonProperties = getButtonProperties();
```

Comment:
I think it can be replaced with useMemo as for the `TaskActionButton `
Also, it looks more like a status, not an actual button, so maybe think about renaming it.